lib/ohai/mixin/azure_metadata.rb

```diff
@@ -22,7 +22,7 @@ module Mixin
     module AzureMetadata
 
       AZURE_METADATA_ADDR = "169.254.169.254" unless defined?(AZURE_METADATA_ADDR)
-      AZURE_METADATA_URL = "/metadata/instance?api-version=2017-04-02" unless defined?(AZURE_METADATA_URL)
+      AZURE_METADATA_URL = "/metadata/instance?api-version=2017-12-01" unless defined?(AZURE_METADATA_URL)
```

Comment:
In the ec2 metadata resource we have some magic that looks for the latest supported metadata from an array of known supported versions. Do you think we should maybe look at doing something like that here or are we guaranteed to always have 2017-12-01?

Reply:
I looked at the documentation and it seems that they support older versions as they age. I'll have to look into seeing if there's a per instance api version checker. Maybe I can bug someone in Azure. I found 12-01 isn't supported in UK so I backed it off to the one that's supported everywhere.